src/main/java/uk/gov/ch/developer/docs/controller/developer/SignOutController.java

```diff
@@ -26,32 +27,21 @@
     private IIdentityProvider identityProviders;
     @Autowired
     private Store store;
+    @Autowired
+    private Oauth2 oauth2;
+
+    public SignOutController() {
+    }
 
     @GetMapping("${signout.url}")
     public void doSignOut(final HttpServletResponse httpServletResponse,
-            final HttpServletRequest httpServletRequest) throws IOException {
+                          final HttpServletRequest httpServletRequest) throws IOException {
 
         LOGGER.debug("Processing sign out");
 
         final Session chSession = (Session) httpServletRequest
                 .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
-
-        if (chSession.getSignInInfo().isSignedIn()) {
-
-            final Map<String, Object> sessionData = chSession.getData();
-            final Map<String, Object> signInInfo =
-                    (Map<String, Object>) sessionData.get(SIGN_IN_INFO);
-
-            signInInfo.replace(SessionKeys.SIGNED_IN.getKey(), 1, 0);
-            sessionData.remove(SIGN_IN_INFO);
-
-            final String zxsKey = (String) sessionData.get(".zxs_key");// This is the id of cookie
-                                                                       // stored in redis
-            if (zxsKey != null) {
-                LOGGER.trace("Deleting ZXS info from cache");
-                store.delete(zxsKey);
-            }
-        }
+        oauth2.invalidateSession(chSession, SIGN_IN_INFO, store);
```

Comment:
Why not take SIGN_IN_INFO and store into the oauth2 stuff so that we don't need to reference them and setup the store bean?

Reply:
Done

Reply:
Correction. The SIGN_IN_INFO has been moved. It was not happy about moving the store

Reply:
What was the problem?

Reply:
Did you try to move it as a spring bean?